src/extension/background-script/actions/nostr/setPrivateKey.ts

```diff
@@ -1,10 +1,32 @@
+import { encryptData } from "~/common/lib/crypto";
 import type { MessagePrivateKeySet } from "~/types";
 
 import state from "../../state";
 
 const setPrivateKey = async (message: MessagePrivateKeySet) => {
-  await state.getState().getNostr().setPrivateKey(message.args.privateKey);
-  return {};
+  const id = message.args?.id || state.getState().currentAccountId;
+
+  const password = state.getState().password as string;
+  const privateKey = message.args.privateKey;
+  const accounts = state.getState().accounts;
+
+  if (id && Object.keys(accounts).includes(id)) {
+    const account = accounts[id];
+    account.nostrPrivateKey = privateKey
+      ? encryptData(privateKey, password)
+      : null;
```

Comment:
I think we should do a special "delete" action to remove the key and not do this by setting the private key to null. I would think that having a special key would prevent the user a bit better from accidentially deleting it.